gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/spring/GraviteeJavaMailSenderImpl.java

```diff
@@ -114,4 +140,28 @@ private Properties loadProperties() {
 
         return properties;
     }
+
+    class ReferenceContext {
+        private String referenceId;
+        private ParameterReferenceType referenceType;
```

Comment:
Can you enrich `GraviteeContext.getCurrentContext()` ?
to have a String field and an enum field for referenceType ?

i think it is sad to dupplicate the work here and I would prefer a more complete object in GraviteeContext.

wdyt ?